aries_vcx/src/protocols/issuance_v2/holder/states/complete.rs

```diff
@@ -0,0 +1,23 @@
+use std::marker::PhantomData;
+
+use messages::msg_fields::protocols::cred_issuance::v2::ack::AckCredentialV2;
+
+use crate::protocols::issuance_v2::formats::holder::HolderCredentialIssuanceFormat;
+
+pub struct Complete<T: HolderCredentialIssuanceFormat> {
```

Comment:
The complete state contains radically different data than `CredentialReceived`, but in practice they are kinda-sorta the same state, obviously this is just marking that we have sent the final ack. Which makes `CredentialReceived` against kind of ephemeral. Which makes me (again) wonder if the 2 states shall not be merged, as typically you would:
- receive `issue-credential` message (being in `OfferPrepared` stated)
- ephemerally transition to `CredentialReceived` state, as most likely right after, you would send `Ack` and transition to `Complete`

But if the 2 states remain separate, I would expect them to hold the same data.

Reply:
Part of the reason i have them separate, is because i started off implementing issue-credential-v2 as the V2.2 impl (i.e. what is seen on main branch of the RFC: https://github.com/hyperledger/aries-rfcs/blob/main/features/0453-issue-credential-v2/README.md). This is bcus in V2.1+ of the issue-credential-v2 protocol, the `CredentialReceived` state is not necessarily the end, if there are MULTIPLE credential being offered, then the holder can reply to an issue-credential message with ANOTHER request-credential message (i.e. transitioning from `CredentialReceived` -> `RequestPrepared` in our state machine).

So i kept them as seperate to support that transition logic. however since then i reverted to just the plain old 2.0 implementation (no support for multi credential issuance). Now the `Completed` state is essentially just a "Ack optionally prepared" state. I like having them seperated personally.. but we can discuss further if you prefer them merged.

> But if the 2 states remain separate, I would expect them to hold the same data.

i.e. the `Completed` state should have a
```
    stored_credential_metadata: T::StoredCredentialMetadata,
```
field?